src/Constraint/BoundsProvidingInterface.php

```diff
@@ -0,0 +1,31 @@
+<?php
+
+/*
+ * This file is part of composer/semver.
+ *
+ * (c) Composer <https://github.com/composer>
+ *
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ */
+
+namespace Composer\Semver\Constraint;
+
+interface BoundsProvidingInterface
```

Comment:
I'm not really sure what the purpose of this interface is?

Reply:
To make sure the constraint provides the new method without breaking BC. We'll also need to implement this on the `MultiConstraint`.

Reply:
I'm still unclear why we would need this? Shouldn't this just be a part of the constraint interface?

Reply:
The point is BC purity as adding a method to the interface breaks implementors, but I'd say we can bump to 2.0 instead.

Reply:
I'm fine with that, I only added it because I thought we want to stay BC. Will remove the interface then.